### PR TITLE
Use Code Action provider based on option and provider

### PR DESCRIPTION
**Pull request by fendor**
Based on the discussions with @alanz and @mpickering

### CODE REVIEW

**Comment by alanz** [comment]
LGTM

File: src/Language/Haskell/LSP/Core.hs [hunk 1]
```diff
@@ -844,6 +844,9 @@ initializeRequestHandler' onStartup mHandler tvarCtx req@(J.RequestMessage _ ori
           static (Just d) _ = Just d
           static _ (Just _) = Just (J.GotoOptionsStatic True)
           static _ Nothing  = Nothing
+          static' (Just d) (Just _) = Just d
```
Comment (jneira) [inline src/Language/Haskell/LSP/Core.hs]: This one is already covered by `static (Just d) _ = Just d`, no?

File: src/Language/Haskell/LSP/Core.hs [hunk 2]
```diff
@@ -844,6 +844,10 @@ initializeRequestHandler' onStartup mHandler tvarCtx req@(J.RequestMessage _ ori
           static (Just d) _ = Just d
           static _ (Just _) = Just (J.GotoOptionsStatic True)
           static _ Nothing  = Nothing
+          
+          static' (Just d) (Just _) = Just d
+          static' _ (Just _) = Just (J.CodeActionOptionsStatic True)
+          static' _ _  = Nothing
```
Comment (mpickering) [inline src/Language/Haskell/LSP/Core.hs]: You could modify `static` to take an extra default argument, then you can use it for both the cases.

Comment (alanz) [reply src/Language/Haskell/LSP/Core.hs]: I am happy with either option.

Comment (fendor) [reply src/Language/Haskell/LSP/Core.hs]: Cases are not identical, if I see that correctly.
`static'` only returns `Just d` if both provider and handler are set, while `static` only requires the provider?

Comment (mpickering) [reply src/Language/Haskell/LSP/Core.hs]: Yeah, but why are they different?

**Comment by jneira** [comment]
oh, sorry, i misreaded the functions names and i thought my comment had no sense anymore, but yeah, separating them (or given them different meaningful names) could help to avoid the mistake

**Pull request merged**